### ISSUE TITLE
No `.m.rule.contains_user_name` rule sent to clients

### ISSUE THREAD

**Issue by deepbluev7** [matrix-org/synapse#14348]
### Description

When I curl `/_matrix/client/v3/pushrules/` it doesn't return the `.m.rule.contains_user_name` rule. This means clients don't properly highlight those messages. It also isn't present in the pseudo account data event

### Steps to reproduce

- curl `/_matrix/client/v3/pushrules/`  and grep for user_name.

### Homeserver

nheko.im, neko.dev, ocean.joedonofry.com

### Synapse Version

1.71rc1, 1.69.0

### Installation Method

Other (please mention below)

### Platform

Docker, custom ebuild, etc

### Relevant log output

```shell
None
```

### Anything else that would be useful to know?

The rule is present in https://github.com/matrix-org/synapse/blob/2d0ba3f89aaf9545d81c4027500e543ec70b68a6/rust/src/push/base_rules.rs, which would suggest it is formatted. The formatting should happen here: https://github.com/matrix-org/synapse/blob/99a7e7e0230cba5d00ec204926edae89d4b6b8c3/synapse/push/clientformat.py

I have no idea yet, where it goes poof.

**Comment by DMRobertson** [comment]
AFAICT from [the spec,](https://spec.matrix.org/v1.4/client-server-api/#default-content-rules) this rule is supposed to show up under the global/content.

This doesn't look promising: 

```
$ curl -s -H "Authorization: Bearer $TOKEN" -X GET https://matrix-client.matrix.org/_matrix/client/v3/pushrules/ | jq '.global.content'
[]
```

**Comment by DMRobertson** [comment]
On 1.68.0, this snippet

https://github.com/matrix-org/synapse/blob/5442891cbca67d3af27c448791589e0b9abeb7f8/synapse/push/clientformat.py#L42-L43

sets `template_rule` to `None` where `r` is

```
PushRule(rule_id='global/content/.m.rule.contains_user_name', priority_class=4, conditions=[{'kind': 'event_match', 'key': 'content.body', 'pattern_type': 'user_localpart'}], actions=['notify', {'set_tweak': 'sound', 'value': 'default'}, {'set_tweak': 'highlight'}], default=True, default_enabled=True)
```

Not a regression then?

**Comment by DMRobertson** [comment]
We return `None` here https://github.com/matrix-org/synapse/blob/5442891cbca67d3af27c448791589e0b9abeb7f8/synapse/push/clientformat.py#L97-L98 because `rule.conditions[0]` is `{'kind': 'event_match', 'key': 'content.body', 'pattern_type': 'user_localpart'}`.

**Comment by DMRobertson** [comment]
@erikjohnston I think you're best-placed to look at this when you're back. I don't fully grok the templaterule machinery here so wouldn't want to attempt a fix without your insight, at the very least.

**Comment by deepbluev7** [comment]
I made a temporary fix in #14356. It is more of a bodge though, so you might want to fix it properly instead.
